core/server/master/src/main/java/alluxio/master/FaultTolerantAlluxioMasterProcess.java

```diff
@@ -78,7 +83,11 @@ public void start() throws Exception {
       mLeaderSelector.waitForState(State.PRIMARY);
       if (gainPrimacy()) {
         mLeaderSelector.waitForState(State.SECONDARY);
-        losePrimacy();
+        if (mRunning) { // To identify whether stop is called
```

Comment:
we can short circuit this:
```java
if (mRunning) {
  break;
}
losePrimacy();
```

Should we also check after `mLeaderSelector.waitForState(State.PRIMARY)`?

Reply:
short circuit it and also check for primary state.